#!/bin/bash
# This script lists the commands to reproduce the key analyses done in the manuscript
# The a minimal set of data to reproduce the analyses is provided
# Random numbers generated may be different
# The exact results generated by the authors are provided in Dryad repository (see GitHub site's description file)

# Part 1. Fitting pulsed evolution models and preparing reference for prediction
# Starting data:  Reference/reference.tre
#                 Reference/16S_GCN.txt
# Expected output:  Reference/homogeneous_data.RDS
#                   Reference/homogeneous_models.RDS
# Fit homogeneous BM and PE models
Rscript fit_homogeneous_models.R Reference/reference.tre Reference/16S_GCN.txt Reference/homogeneous
# Binary partition of the phylogeny
Rscript binary_partition_by_AIC.R Reference/homogeneous_data.RDS Reference/homogeneous_models.RDS Reference/binary_partition.RDS Reference/rescaled_data_model.RDS
# Prepare prediction reference
Rscript prepare_prediction_data.R Reference/rescaled_data_model.RDS Reference/prepared_reference.RDS

# Part 2. Cross-validation at different NSTD cutoff
# Starting data: results from Part 1.
# Expected output:  CV/GCN.PE.CV.RDS
                    CV/GCN.BM.CV.RDS
                    CV/GCN.MP_EMP.CV.RDS
# Part 3. Predicting GCN for SILVA and HMP1 

# Part 4. Microbiome analyses with GCN correction

# Part 5. Making figures and Tables
# Figure 1 and 2 
# Starting data:
# Expected output:  Fig_1.png Fig_1.pdf
Rscript Figure_1_Expectation.R
Rscript Figure_2_GCN_Classification.R

# Figure 3 and 4
# Starting data:
# Expected output:  
Rscript Figure_3_Abundance.R
Rscript Figure_4_Beta_diversity_Bray.R
Rscript Figure_S1.R
Rscript Figure_S2_Reference_insertion.R
Rscript Figure_S3_distance_between_taxonomic_group.R
#

